src/capwap_config.erl

```diff
@@ -261,3 +231,15 @@ update_wlan_config(RadioId, WlanId, Settings, #wtp{radios = Radios} = Config) ->
 						    '#set-'(Settings, WLAN))},
     Config#wtp{radios = lists:keystore(RadioId, #wtp_radio.radio_id,
 				       Radios, Radio1)}.
+
+merge(Settings, DefValues) ->
```

Comment:
regarding a), seem I introduced the inefficient version, but still...

I also don't see why you needed to modify the original wtp_get logic in that way.

The HTTP API does not need a defaults mechanisms. Any type of templateing would be handled by the server side, not the AC side. So we can assume that we always get a consistent and complete configuration of that API.
This might make is necessary to forward some (or all) information that we get during the handshake to the server.

The default overrides are only needed for the local/static configuration.

@mgumz comments???

Reply:
regarding performance: until its measured i dont see the need to pre-optimized the current working  way of handling wtp-attachment to a supposed-to-be-faster method for (in the current situation) 1k devices for an event which is rather rare. if profiling turns out that this is a slow-path-bottleneck: yes, then we can optimize it later on

getting a full-fledged 100% config from the outside and no-need-to-merge anything: yeah, thats the endgoal. but the merging of the external per-wtp-config with the base set of the default-configuration seems like a valid option to me. if the provided external-config lacks values (eg when the capwap-ac has learned a new trick and the external-configs are not "updated" yet up to 100%) one needs the fallthrough anyway.

for me the merge is not a show-stopper at all, neither from a performance perspective nor from a this-is-how-it-works perspective.